HPCC-13381 Add #option ('obfuscateOutput') to hide original ECL

ghalliday opened the pull request:
Useful if you wish to publish a query to a third part, but hide details
of the ECL implementation.

Signed-off-by: Gavin Halliday gavin.halliday@lexisnexis.com

Michael-Gardner committed:
Created cleanupRuntimeEnvironment

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

Michael-Gardner committed:
HPCC-12852 Modified more of configmgr.in

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

Michael-Gardner committed:
HPCC-12978 Stop_thor uses kill_process.  kill_process parameter order changed

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

Michael-Gardner committed:
HPCC-13058 Check frunssh failure in run_thor

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

Commit:
HPCC-12965 Fix xsl:message handling

	- Handle terminate yes/no properly
	- Cleanup user error messages a bit

Signed-off-by: Gleb Aronsky <gleb.aronsky@lexisnexis.com>

Michael-Gardner committed:
HPCC-12978 Stop_thor uses kill_process.  kill_process parameter order changed

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

richardkchapman committed:
Split off candidate-5.4.0 branch

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

Michael-Gardner committed:
HPCC-12978 Made changes in line with PR

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

richardkchapman committed:
Merge pull request #7062 from garonsky/HPCC-12965_fix_xsl_message_5_4

HPCC-12965 Fix xsl:message handling

Reviewed By: Kevin Wang <kevin.wang@lexisnexis.com>
Reviewed By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7085 from Michael-Gardner/HPCC-13058-5.4

HPCC-13058 Frunssh exception caught in run_thor

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-5.2.0' into candidate-5.4.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-5.2.0' into candidate-5.4.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

jakesmith committed:
HPCC-13207 Avoid uint overflow and crash if lookup HT huge.

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

richardkchapman committed:
Merge pull request #7104 from Michael-Gardner/HPCC-12852-prfix

HPCC-12852 Configmgr PR fix

Reviewed-By: Xiaoming Wang <xiaoming.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #7098 from jakesmith/hpcc-13207

HPCC-13207 Avoid uint overflow and crash if lookup HT huge.

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
HPCC-13229 Improve error reporting of #WEBSERVICE and other #commands

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
HPCC-13266 Improve processing of global scalar expressions

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

wangkx committed:
HPCC-13113 Fix incorrect use of default_client_version

Inside ESP ecm files, the 'version' variable defines the
latest version of an ESP service. The 'default_client_
version' variable may be used to define the version of an
HTTP ESP client when an ESP HTTP client does not specify
its version. In this fix, the 'default_client_version'
variables are removed from the existing ECLWatch ESP
services since, until now, ECLWatch HTTP forms always use
the latest version of an ESP service and the 'version'
variables have already been defined in the ECLWatch ESP
services. (The 'default_client_version' variable may added
whenever needed later.) The hidl code is modified to use
the 'version' variable as the version of an HTTP client
if the client does not specify its own version and the
'default_client_version' variable is not defined.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

jakesmith committed:
HPCC-13239 Release stopped distribute buckets as soon as possible

If some slaves stop early, other slaves that have partially built
up target buffers to those stopped slaves need to release those
buckets, allowing the rest to fully utilize the available shared
send buffer.

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

richardkchapman committed:
Merge pull request #7112 from Michael-Gardner/hpcc-12978-rebase

HPCC-12978 PR fix for kill_process

Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-5.2.0' into candidate-5.4.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

jakesmith committed:
HPCC-13247 Delete temporary spill file from smart buffer

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

ghalliday committed:
HPCC-13249 Do not create temporary expressions for events

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
HPCC-13245 Improve temporary row code for no_selectnth

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
HPCC-13246 Constant rows are now legal to link count

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

jakesmith committed:
HPCC-13239 Minor changes following review

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

jakesmith committed:
HPCC-13239 One more minor change.

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

Commit:
HPCC-13251 Fix HPCC Windows VS 9 2008 Win64 errors

ghalliday committed:
Merge pull request #7121 from jakesmith/hpcc-13239

HPCC-13239 Release stopped distribute buckets as soon as possible

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #7123 from jakesmith/hpcc-13247

HPCC-13247 Delete temporary spill file from smart buffer

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

Commit:
HPCC-13251 Update configuration for Win32

The changes will make Win32 and Win64 use the same level dependent libraries.
Remove add_definitions(/Zc:wchar_t-) which defines XMLCh as "unsigned short"

For x64 debug build need copy "C:\Program Files(x86)\Microsoft Visual Studio 9.0\VC\redist\Debug_NonRedist\amd64\Microsoft.VC90.DebugCRT\*.* to <HPCC Build>/bin/Debug/ then rebuild from Visual Studio 9 2008 (Debug x64).

Remove MTd
remove __WIN32 and reset CPPUNIT and MYSQL INCLUDE PATH
add Win64 MySQL support on Windows

jakesmith committed:
HPCC-13272 Fix to allow localThor setup to work with hostnames

With a local Thor setup (localThor=true), if hostnames were used
in the environmnent as opposed to "." or IP's, Thor failed to
start for a couple of reasons.
a) the daliadmin command lines in the script expected "." or IP's.
Fixed with addition of server=
b) the hostname failed to match the IP's in the thorgroup with
the consequence that no slaves were started by the start_slaves
scripts. Fixed by using an IP from the group.

Signed-off-by: Jake Smith <jake.smith@lexisnexis.com>

Michael-Gardner committed:
HPCC-13278 Now remove Defaults:hpcc line from sudoers file

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

afishbeck committed:
HPCC-13280 Fix incomplete foreign LFN for roxie cache and remote lookup

Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

richardkchapman committed:
Merge pull request #7125 from ghalliday/issue13249

HPCC-13249 Do not create temporary expressions for events

Reviewed By: Jamie Noss <james.noss@lexisnexis.com>
Reviewed By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7128 from ghalliday/issue13245

HPCC-13245 Improve temporary row code for no_selectnth

Reviewed By: Jamie Noss <james.noss@lexisnexis.com>
Reviewed By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7111 from wangkx/h13113_540

HPCC-13113 Fix incorrect use of default_client_version

Reviewed By: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>
Reviewed By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7142 from Michael-Gardner/HPCC-13278

HPCC-13278 Now remove Defaults:hpcc line from sudoers file

Reviewed By: Xiaoming Wang <xiaoming.wang@lexisnexis.com>
Reviewed By: Richard Chapman <rchapman@hpccsystems.com>

Michael-Gardner committed:
HPCC-13279 Added removal of remote_install.tgz

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

Michael-Gardner committed:
HPCC-12987 Added case statement for new thor exit codes

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

richardkchapman committed:
Merge pull request #7144 from Michael-Gardner/HPCC-13279

HPCC-13279 Added removal of remote_install.tgz

Reviewed-By: Xiaoming Wang <xiaoming.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7124 from ghalliday/issue13229

HPCC-13229 Improve error reporting of #WEBSERVICE and other #commands

Reviewed-By: Jamie Noss <james.noss@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7126 from ghalliday/issue13246

HPCC-13246 Constant rows are now legal to link count

Reviewed-By: Jamie Noss <james.noss@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

mckellyln committed:
HPCC-13289 Cluster_script.py trap ctrl-c and exit

Signed-off-by: Mark Kelly <mark.kelly@lexisnexis.com>

richardkchapman committed:
Merge branch 'candidate-5.2.0' into candidate-5.4.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7152 from mckellyln/pythonabrt

HPCC-13289 Cluster_script.py trap ctrl-c and exit

Reviewed By: Xiaoming Wang <xiaoming.wang@lexisnexis.com>
Reviewed By: Michael Gardner <michael.gardner@lexisnexis.com>
Reviewed By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7136 from ghalliday/globalcse

HPCC-13266 Improve processing of global scalar expressions

Reviewed-By: Jamie Noss <james.noss@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7143 from afishbeck/roxieForeignFilePath

HPCC-13280 Fix incomplete foreign LFN for roxie cache and remote lookup

Reviewed By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7141 from jakesmith/hpcc-13272

HPCC-13272 Fix to allow localThor setup to work with hostnames

Reviewed-By: Michael Gardner <michael.gardner@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

afishbeck committed:
HPCC-13002 Allow roxie target to be specified in the url

If the same query exists in multiple targets on the same roxie process
the URL can be used to disambiguate.

Example: http://host:9876/roxie1?.trim=0

Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

afishbeck committed:
HPCC-13002 Update based on review

Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

wangkx committed:
HPCC-13220 Remove eclagent from preflight report

When running a preflight, both eclagent and agentexec are
reported by the existing ESP preflight function. Since
eclagent is an on-demand procedure, this fix removes it
from ESP preflight report. So, it will not be displayed
on the ECLWatch preflight page.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

jamienoss committed:
HPCC-12462 Share usage info between eclcc and eclcmd

This has two accompanying changes:
 * --debug (synonym for -g) was added to eclcc as it is so for eclcmd.
 * eclcmd now checks for verbosity after finding --help.
   There was no verbose section of the usage info prior, which is
   why it made no sense to check i.e. '-v' wasn't related to '--help'

Signed-off-by: James Noss <james.noss@lexisnexis.com>

Michael-Gardner committed:
HPCC-13301 Added python to ubuntu/debian releases as dependency

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

richardkchapman committed:
Merge pull request #7168 from wangkx/h13220

HPCC-13220 Remove eclagent from preflight report

Reviewed-By: Russ Whitehead <william.whitehead@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-5.2.2' into candidate-5.4.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7176 from Michael-Gardner/hpcc-13301

HPCC-13301 Added python to ubuntu/debian releases as dependency

Reviewed-By: Xiaoming Wang <xiaoming.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

Michael-Gardner committed:
HPCC-13335 Suppressed output from kill and rm commands

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

richardkchapman committed:
Split off candidate-5.2.4 branch

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-5.2.4' into candidate-5.4.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

Michael-Gardner committed:
HPCC-13285 Install cluster can pass previously generated keys

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

ghalliday committed:
HPCC-13357 Improve error message for invalid ROWDIFF

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

afishbeck committed:
HPCC-13002 Further update based on review

Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

ghalliday committed:
Merge pull request #7156 from afishbeck/roxieHttpTarget

HPCC-13002 Allow roxie target to be specified in the url

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge pull request #7137 from xwang2713/HPCC_13251

HPCC-13251 Fix HPCC Windows VS 9 2008 Win64 errors

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

ghalliday committed:
Merge remote-tracking branch 'origin/candidate-5.2.2' into candidate-5.4.0

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday committed:
Merge remote-tracking branch 'origin/candidate-5.2.2' into candidate-5.4.0

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

Conflicts:
	thorlcr/activities/hashdistrib/thhashdistribslave.cpp

ghalliday committed:
HPCC-13381 Add #option ('obfuscateOutput') to hide original ECL

Useful if you wish to publish a query to a third part, but hide details
of the ECL implementation.

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

ghalliday commented:
@jamienoss please review.  (Compare with ?w=1 to simplify whitespace changes)

ghalliday committed:
HPCC-13081 Ignore ,MERGE on non-grouping Aggregate

Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

Commit:
HPCC-11665 Banner to support color picker

Allow the admin to set the color for the text that scrolls along the marquee using dojo's color picker module.

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

jamienoss commented on ecl/eclcc/eclcc.cpp:
```diff
@@ -2248,6 +2248,7 @@ const char * const helpText[] = {
     "! -fexportDependencies    Generate information about inter-definition dependencies",
     "! -fmaxCompileThreads     Number of compiler instances to compile the c++",
     "! -fnoteRecordSizeInGraph Add estimates of record sizes to the graph",
+    "! -fobfuscateOutput       Remove details of the original ECL from output",
```
Shouldn't -fshowEclInGraph also be listed?

ghalliday replied:
I don't think it is significant enough to include.  This list doesn't include all options - only those that are possibly useful.

jamienoss replied:
https://track.hpccsystems.com/browse/HPCC-12313

richardkchapman committed:
Merge pull request #7207 from miguelvazq/HPCC-11665

HPCC-11665 Banner to support color picker

Reviewed-By: Gordon Smith <gordon.smith@lexisnexis.com> 
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7196 from ghalliday/issue13357

HPCC-13357 Improve error message for invalid ROWDIFF

Reviewed-By: Jamie Noss <james.noss@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7181 from Michael-Gardner/HPCC-13335

HPCC-13335 Suppressed output from kill and rm commands

Reviewed-By: Xiaoming Wang <xiaoming.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7189 from Michael-Gardner/HPCC-13285

HPCC-13285 Install cluster can pass previously generated keys

Reviewed-By: Xiaoming Wang <xiaoming.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7204 from ghalliday/issue13081

HPCC-13081 Ignore ,MERGE on non-grouping Aggregate

Reviewed-By: Jamie Noss <james.noss@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

Michael-Gardner committed:
HPCC-13401 Removed unused default passwords from Host class

Signed-off-by: Michael Gardner <Michael.Gardner@lexisnexis.com>

ghalliday committed:
Merge pull request #7173 from jamienoss/issue12462-eclcmd-eclcc-usage

HPCC-12462 Share usage info between eclcc and eclcmd

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman committed:
HPCC-13433 Misleading tracing on Roxie startup

Suppress the tracing from failure to locate workunit in cases when it's
expected to fail.

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

ghalliday committed:
Merge pull request #7228 from richardkchapman/failed-to-locate-symbol

HPCC-13433 Misleading tracing on Roxie startup

Reviewed-by: Gavin Halliday <ghalliday@hpccsystems.com>

richardkchapman commented:
This is a new feature and should not be included in a point release. Please retarget to 5.4

richardkchapman committed:
Merge pull request #7223 from Michael-Gardner/hpcc-13401

HPCC-13401 Host.py contains default passwords that should be removed

Reviewed-By: Xiaoming Wang <xiaoming.wang@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge branch 'candidate-5.2.2' into candidate-5.4.0

Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

richardkchapman committed:
Merge pull request #7183 from Michael-Gardner/hpcc-12987

HPCC-12987 Added case statement for new thor exit codes

Reviewed-By: Jake Smith <jake.smith@lexisnexis.com>
Reviewed-By: Richard Chapman <rchapman@hpccsystems.com>

ghalliday commented:
Closing - see https://github.com/hpcc-systems/HPCC-Platform/pull/7238 instead
